extensions/positron-python/src/client/common/installer/uvInstaller.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/* eslint-disable class-methods-use-this */
+
+import { inject, injectable } from 'inversify';
+import { ModuleInstallerType } from '../../pythonEnvironments/info';
+import { ExecutionInfo, IConfigurationService } from '../types';
+import { ModuleInstaller } from './moduleInstaller';
+import { InterpreterUri, ModuleInstallFlags } from './types';
+import { isUvInstalled } from '../../pythonEnvironments/common/environmentManagers/uv';
+import { IServiceContainer } from '../../ioc/types';
+import { isResource } from '../utils/misc';
+import { IWorkspaceService } from '../application/types';
+import { IInterpreterService } from '../../interpreter/contracts';
+
+@injectable()
+export class UVInstaller extends ModuleInstaller {
+    constructor(@inject(IServiceContainer) serviceContainer: IServiceContainer) {
+        super(serviceContainer);
+    }
+
+    public get name(): string {
+        return 'Uv';
+    }
+
+    public get displayName(): string {
+        return 'uv';
+    }
+
+    public get type(): ModuleInstallerType {
+        return ModuleInstallerType.Uv;
+    }
+
+    public get priority(): number {
+        return 30;
+    }
+
+    public async isSupported(_resource?: InterpreterUri): Promise<boolean> {
+        // uv can be used in any environment type
+        try {
+            return await isUvInstalled();
+        } catch {
+            return false;
+        }
+    }
+
+    protected async getExecutionInfo(
+        moduleName: string,
+        resource?: InterpreterUri,
+        flags: ModuleInstallFlags = 0,
+    ): Promise<ExecutionInfo> {
+        // If the resource isSupported, then the uv binary exists
+        const execPath = 'uv';
+        // TODO: should we use uv add if a pyproject.toml exists?
+        const args = ['pip', 'install', '--upgrade'];
+
+        // Get the path to the python interpreter (same as ModuleInstaller install code)
```

Comment:
nit, if this is all the same, can we call super()?

Reply:
ah I should clarify this - this code block is similar to a small part in `ModuleInstaller.installModule()`. I think it would be more of a headache to figure out how to DRY it lol